discovery_engine_core/web-api/src/db.rs

```diff
@@ -12,82 +12,48 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-use serde_json::from_reader;
-use std::{collections::HashMap, fs::File, path::PathBuf, sync::Arc};
+use std::sync::Arc;
 use xayn_discovery_engine_ai::{CoiConfig, CoiSystem, GenericError};
-use xayn_discovery_engine_bert::{AveragePooler, SMBert, SMBertConfig};
-use xayn_discovery_engine_tokenizer::{AccentChars, CaseChars};
 
-use crate::{
-    elastic,
-    models::{IngestedDocument, PersonalizedDocument},
-    storage::UserState,
-};
+use crate::{elastic, elastic::ElasticState, storage::UserState};
 
 pub(crate) type Db = Arc<AppState>;
 
-#[allow(dead_code)]
 pub(crate) struct AppState {
-    pub(crate) smbert: SMBert,
+    /// The center of interest (coi) system.
     pub(crate) coi: CoiSystem,
-    pub(crate) documents_by_id: HashMap<String, PersonalizedDocument>,
-    pub(crate) documents: Vec<PersonalizedDocument>,
+    /// Elastic Search client.
+    pub(crate) elastic_state: ElasticState,
+    /// Handler for storing the user state.
     pub(crate) user_state: UserState,
 }
 
 impl AppState {
-    fn new(
-        documents_by_id: HashMap<String, PersonalizedDocument>,
-        smbert: SMBert,
-        user_state: UserState,
-    ) -> Self {
-        let documents = documents_by_id.clone().into_values().collect();
+    fn new(elastic_state: ElasticState, user_state: UserState) -> Self {
         Self {
-            documents_by_id,
-            documents,
-            smbert,
             coi: CoiConfig::default().build(),
+            elastic_state,
             user_state,
         }
     }
 }
 
 #[derive(Clone, Debug)]
 pub(crate) struct InitConfig {
-    /// S-mBert vocabulary path.
-    pub(crate) smbert_vocab: PathBuf,
-    /// S-mBert model path.
-    pub(crate) smbert_model: PathBuf,
-    /// List of [IngestedDocument]s in JSON format.
-    pub(crate) data_store: PathBuf,
-    /// Handler for storing the user state.
-    pub(crate) user_state: UserState,
+    /// PostgreSQL url
+    pub(crate) pg_url: String,
     /// Elastic configuration.
     #[allow(dead_code)]
     pub(crate) elastic: elastic::Config,
 }
 
 // NOTE this will be removed by follow up tasks so it's not necessary to validate data here anymore
-pub(crate) fn init_db(config: &InitConfig) -> Result<Db, GenericError> {
-    let smbert = SMBertConfig::from_files(&config.smbert_vocab, &config.smbert_model)?
-        .with_accents(AccentChars::Cleanse)
-        .with_case(CaseChars::Lower)
-        .with_pooling::<AveragePooler>()
-        .with_token_size(64)?
-        .build()?;
+pub(crate) async fn init_db(config: InitConfig) -> Result<Db, GenericError> {
```

Comment:
:art: this could be made a constructor function of `AppState` directly
```suggestion
impl AppState {
    pub(crate) async fn init_db(config: InitConfig) -> Result<Arc<Self>, GenericError> {
```